viewing_party/party.py

```diff
@@ -1,23 +1,138 @@
+from re import T
+
 # ------------- WAVE 1 --------------------
 
+# This function adds 3 key-value pairs to a dictionary called movie_dict
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title and genre and rating:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+    else:
+        return None
+
+# This function adds a key-value pair called "watched" to a dictionary called user_data; It includes all movies that the user has watched 
+def add_to_watched(user_data, movie):
+    user_data["watched"] = [movie]
+    return user_data
+
+# This function adds a key-value pair called "watchlist" to a dictionary called user_data. It includes all movies that the user wants to watch 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = [movie]
+    return user_data
+
+# This function decides whether to put a movie into a watchlist or a list of watchd movie
+def watch_movie(user_data, title):
+    for index in range(len(user_data["watchlist"])):
+        if title == user_data["watchlist"][index]["title"]:
+            watched_movie = user_data["watchlist"][index]
+            user_data["watchlist"].remove(watched_movie)
+            user_data["watched"].append(watched_movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# This function calculates the average rating of all moviws in the watched movies
+def get_watched_avg_rating(user_data):
+    average_rating = 0
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    else:
+        for movie_watched in user_data["watched"]:
+            average_rating += movie_watched["rating"]
+        return average_rating / len(user_data["watched"])
+
+# This function finds out the genre that is the most fewquently watched 
+def get_most_watched_genre(user_data):
+    # example of dict_genre_count: {'Fantasy': 3, 'Action': 1, 'Intrigue': 2}
+    dict_genre_count = {}
+    if len(user_data["watched"]) == 0:
+        return None
+    else:
+        for movie_watched in user_data["watched"]:
+            if movie_watched["genre"] not in dict_genre_count.keys():
+                dict_genre_count[movie_watched["genre"]] = 1
+            else:
+                dict_genre_count[movie_watched["genre"]] += 1

```

Comment:
You can clean this up a little using the [`.get`](https://docs.python.org/3/library/stdtypes.html#dict.get) dictionary method:

```suggestion
        for movie_watched in user_data["watched"]:
            genre = movie_watched["genre"]
            dict_genre_count[genre] = dict_genre_count.get(genre, 0) + 1
```